main/src/library/File.flix

```diff
@@ -59,4 +159,180 @@ namespace File {
+            case ex: ##java.io.IOException =>
+                import java.lang.Throwable.getMessage();
+                Iterator.singleton(Err(getMessage(ex)))
+        }
+
+    ///
+    /// Returns an array of all the bytes in the given file `f`.
+    ///
+    pub def readBytes(f: String): Result[Array[Int8], String] & Impure =
+        try {
+            import new java.io.File(String) as newFile;
+            import java.io.File.toPath();
+            import java.nio.file.Files:readAllBytes(##java.nio.file.Path);
+
+            let javaFile = newFile(f);
+            let javaPath = toPath(javaFile);
+            let bytes = readAllBytes(javaPath);
+
+            Ok(bytes)
+        } catch {
+            case ex: ##java.io.IOException =>
+                import java.lang.Throwable.getMessage();
+                Err(getMessage(ex))
+        }
+
+    ///
+    /// Returns an iterator of the bytes in the given `file` in chunks of size `chunkSize`.
+    ///
+    pub def readChunks(f: String, chunkSize: Int32): Iterator[Result[Array[Int8], String]] & Impure =
+        try {
+            import new java.io.File(String) as newFile;
+            import java.io.File.toPath();
+            import java.nio.file.Files:readAllBytes(##java.nio.file.Path);
+
+            let javaFile = newFile(f);
+            let javaPath = toPath(javaFile);
+            let bytes = readAllBytes(javaPath);
+            let byteCount = Array.length(bytes);
+            let i = ref 0;
+
+            let done = () -> deref i >= byteCount;
+            let next = () -> {
+                let j = deref i;
+                if (j < byteCount) {
+                    let end = {
+                        if (j + chunkSize >= byteCount) byteCount - 1
+                        else j + chunkSize - 1
+                    };
+                    let byteOfChuckSize = Array.slice(j, end, bytes);
+                    i := j + chunkSize;
+                    Ok(byteOfChuckSize)
+                } else bug!("Empty Iterator!")
+            };
+            Iterator(done, next)
+        } catch {
+            case ex: ##java.io.IOException =>
+                import java.lang.Throwable.getMessage();
+                Iterator.singleton(Err(getMessage(ex)))
+        }
+
+    ///
+    /// Writes `data` to the given file `f`.
+    ///
+    /// Creates file `f` if it does not exist. Overwrites it if it exists.
+    ///
+    /// Returns `true` if the file was created.
+    ///
+    pub def write(f: String, data: t): Result[Bool, String] & Impure with ToString[t] =
+        try {
+            import new java.io.File(String) as newFile;
+            import new java.io.PrintWriter(##java.io.File) as newPrintWriter;
+            import java.io.PrintWriter.println(String);
+            import java.io.PrintWriter.close();
+            import java.io.PrintWriter.checkError();
+
+            let javaFile = newFile(f);
+            let w = newPrintWriter(javaFile);
+
+            println(w, "${data}");
```

Comment:
println is nice since it uses the system line seperator but I think it would be nice to support the PrintWriter print function too. maybe just a flag on write?

Reply:
I think this is the best impl. You get the system line sep which is almost always what you want. 

What do you mean by a flag?

Reply:
I suppose files should always end with a line separator.

> What do you mean by a flag?

A `newLine: Bool` argument that decides whether to use `println` or `print`. Maybe two public methods on op called `writeln`/`write`. It seemed easy to add and I assumed there was some use case since java had it